### PR TITLE
Postgres: Allow CREATE TABLE INHERITS with no new columns

### PR DESCRIPTION
**Pull request by greg-finley**
### Brief summary of the change made

Closes https://github.com/sqlfluff/sqlfluff/issues/5092

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 1]
```diff
@@ -1749,32 +1749,37 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
...
-                            Sequence(
-                                "LIKE",
-                                Ref("TableReferenceSegment"),
-                                AnyNumberOf(Ref("LikeOptionSegment"), optional=True),
-                            ),
-                        ),
-                    )
+                        )
+                    ),
+                    Bracketed(),  # Sometimes no columns, like if INHERITS
```
Comment (greg-finley) [inline src/sqlfluff/dialects/dialect_postgres.py:1782]: Everything but this line is just reindented under a `OneOf`

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 2]
```diff
@@ -1749,32 +1749,37 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
...
-                                    # A single COLLATE segment can come before or after
-                                    # constraint segments
-                                    OneOf(
-                                        Ref("ColumnConstraintSegment"),
-                                        Sequence(
-                                            "COLLATE",
-                                            Ref("CollationReferenceSegment"),
+                OneOf(
+                    Bracketed(
+                        Delimited(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:1754]: If we stick an `optional=True` here, we can remove the outer `OneOf`, and the empty `Bracketed()` below

Comment (greg-finley) [reply src/sqlfluff/dialects/dialect_postgres.py]: Like this? The tests don't parse in that case: https://github.com/sqlfluff/sqlfluff/pull/5100/commits/397cae6545759f15324e6eae9974eb7c757a29ca

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 3]
```diff
@@ -1749,43 +1749,39 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
...
-                        )
-                    ),
-                    Bracketed(),  # Sometimes no columns, like if INHERITS
+                            Ref("TableConstraintSegment"),
+                            Sequence(
+                                "LIKE",
+                                Ref("TableReferenceSegment"),
+                                AnyNumberOf(Ref("LikeOptionSegment"), optional=True),
+                            ),
+                        ),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:1776]: sorry, i meant the optional to be part of the Delimited() block, inside the Bracketed

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**